contracts/DelFiPrice.sol

```diff
@@ -9,40 +9,60 @@ import "./OpenOracleView.sol";
  * @author Compound Labs, Inc.
  */
 contract DelFiPrice is OpenOracleView {
-    /**
-     * @notice The event emitted when a price is written to storage
-     */
+    /// @notice The event emitted when a price is written to storage
     event Price(string symbol, uint64 price);
 
-    /**
-     * @notice The mapping of medianized prices per symbol
-     */
+    /// @notice The reporter address whose prices checked against the median for safety
+    address anchor;
+
+    /// @notice The highest ratio of the new median price to the anchor price that will still trigger the median price to be updated
+    uint256 upperBoundAnchorRatio;
+
+    /// @notice The lowest ratio of the new median price to the anchor price that will still trigger the median price to be updated
+    uint256 lowerBoundAnchorRatio;
+
+    /// @notice The mapping of medianized prices per symbol
     mapping(string => uint64) public prices;
 
-    constructor(OpenOraclePriceData data_, address[] memory sources_) public OpenOracleView(data_, sources_) {}
+    /**
+     * @param data_ Address of the Oracle Data contract
+     * @param sources_ The reporter addresses whose prices will be used to calculate the median
+     * @param anchor_ The reporter address whose prices checked against the median for safety
+     * @param anchorToleranceMantissa_ The tolerance allowed between the anchor and median. A tolerance of 1e17 means a new median that is 10% off from the anchor will still be saved
+     */
+    constructor(OpenOraclePriceData data_, address[] memory sources_, address anchor_, uint anchorToleranceMantissa_) public OpenOracleView(data_, sources_) {
+        anchor = anchor_;
+        require(anchorToleranceMantissa_ < 1e18, "Anchor Tolerance is too high");
+        upperBoundAnchorRatio = 1e18 + anchorToleranceMantissa_;
```

Comment:
Just to verify, this is basically meant to be `100% + 10%` and `100% - 10%`, right?

Reply:
yes

Reply:
Bit of a nit but I still like the convention for XXe16 for % mantissas, we always get confused about scales and we started to do that in other contracts which I think makes it relatively clear